ViAn/Video/overlay.cpp

```diff
@@ -20,21 +20,46 @@ Overlay::~Overlay() {
 
 /**
  * @brief Overlay::draw_overlay
- * Draws an overlay on top of the specified frame.
+ * Draws the text drawings from an overlay on top of the specified frame.
  * @param img Frame to draw on
  * @param frame_nr Number of the frame currently shown in the video.
  */
 void Overlay::draw_overlay(cv::Mat &frame, int frame_nr) {
     if (show_overlay) {
         for (auto it = overlays[frame_nr].begin(); it != overlays[frame_nr].end(); it++) {
-            if ((*it)->get_show()) {
+            // Only draw the text drawings
+            if ((*it)->get_show() && (*it)->get_shape() == TEXT) {
```

Comment:
Might be more efficient to store the value of `*it` instead of dereferencing twice. Not completely sure though

Reply:
I don't know what's more fficient but this is such a small thing with minor impact on efficiency that readability should be what's most important. The compiler will handle the code and from what I understand will treat the two options the same. I can change it to store the value if we feel like that is more readable.